### PR TITLE
[OSF-7479][OSF-7264] Test and fix some of citeproc-py wrong citations.

### PR DESCRIPTION
**Pull request by hmoco**
## Purpose
Improve the accuracy of citations provided by the api, and create a testing suite to use for future improvements.

## Changes
Addition of a json file populated with citations made using citeproc-js, a test suite comparing citeproc-py to the test file, and some small improvements to our render citation.
With the improvements, 368 citation types differ between our APi and the project page, down from 700+

## Ticket
https://openscience.atlassian.net/browse/OSF-7264

### CODE REVIEW

File: tests/test_citeprocpy.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+from api.citations.utils import render_citation
+
+
+class Node:
+    _id = '2nthu'
+    csl = {'publisher': 'Open Science Framework', 'author': [{'given': u'Henrique', 'family': u'Harman'}], 'URL': 'localhost:5000/2nthu', 'issued': {'date-parts': [[2016, 12, 6]]}, 'title': u'The study of chocolate in its many forms', 'type': 'webpage', 'id': u'2nthu'}
+
+
+class TestCiteprocpy:
+    @unittest.skip
```
Comment (brianjgeiger) [inline tests/test_citeprocpy.py]: Please divide up the json into "tests you expect to pass" and "tests you expect to fail" and have two test methods, one to check each. If a tests you expect to pass fails or vice versa, report a failure, but I'd like these tests to run regardless.

**Comment by brianjgeiger** [comment]
@hmoco Why didn't you just use this for https://openscience.atlassian.net/browse/OSF-7479

**Comment by hmoco** [comment]
@brianjgeiger thats pretty much what I ended up doing, but without the tests. I guess if the tests are also desired, I can solve merge conflicts and link this to the ticket instead

**Comment by brianjgeiger** [comment]
Ah, merge conflicts. Ugh. No, just do a demo for product.

**Comment by brianjgeiger** [comment]
@hmoco `Flake8: ./admin/base/settings/defaults.py:9:40: W291 trailing whitespace`

**Comment by hmoco** [comment]
@brianjgeiger, taking a look at this, not sure how it's failing flake. Passes it locally and that path points to a file that was not touched by this pr. Any ideas?

**Comment by brianjgeiger** [comment]
Maybe develop just had a bad patch right when those commits were pushed? When travis works again we'll restart the tests and see what happens.

**Pull request closed without merging**